lib/buildkite/builder/pipeline.rb

```diff
@@ -41,19 +41,23 @@ def initialize(root, logger: nil)
       end
 
       def upload
+        # Generate the pipeline YAML first.
+        contents = to_yaml
+
+        upload_artifacts
```

Comment:
I think it makes sense to upload artifacts before the pipeline since subsequent steps may rely on these artifacts. The regression was introduced here: https://github.com/Gusto/buildkite-builder/pull/48 ... but I think this gets us what we want.